### DISCUSSION TITLE
Provide a Command to Upgrade Dependencies Similar to pnpm update

### DISCUSSION THREAD

**Discussion by etienne-dldc** [denoland/deno#26211] [Ideas]
Hi,

pnpm offers a [`pnpm update`](https://pnpm.io/cli/update) command that allows users to automatically upgrade dependencies. This functionality is something I rely on heavily, and it’s currently one of the main reasons I haven’t fully switched to using Deno 2 over Node/PNPM.

## Details on the pnpm Command

- I frequently use the --interactive --latest options to selectively upgrade dependencies (pnpm up -Li).
- pnpm also provides an outdated command that can output results in JSON format (pnpm outdated --format json), which is useful for building custom tooling.

## Deno Ecosystem

There are a few community projects that attempt to address this issue:

- [@molt/cli](https://jsr.io/@molt/cli)
- [deno-udd](https://github.com/hayd/deno-udd)
- [@check/deps](https://jsr.io/@check/deps)

The challenge with these user-land solutions is that they often lag behind changes in the Deno ecosystem. For example, after the introduction of JSR support, it took time for these tools to adapt. I’m facing a similar issue now with @molt/cli, which does not yet understand version 4 of the lockfile format.

## Minimal Expected Functionality

At a minimum, I would expect the ability to list outdated dependencies, updating in place could come later...

**Comment by bartlomieju** [comment]
FYI I started working on a prototype for that before Deno 2, but was not able to finish it. I got `deno outdated` working, but not `deno update`. Here's a WIP PR: https://github.com/denoland/deno/pull/26219

**Comment by etienne-dldc** [comment]
The [oudated](https://docs.deno.com/runtime/reference/cli/outdated/) command now does everything I need ! Thanks Deno team !
